src/odemis/acq/stitching/_tiledacq.py

```diff
@@ -1190,3 +1207,198 @@ def run(self):
+
+    # Sort the (selected) centers along X, and for centers with the same X, along the Y.
+    # In theory, the order doesn't really matter (at best it would safe a few
+    # seconds of stage movement), but it's nice for the user that acquisitions
+    # are done always in the same order, as it reduces "astonishment".
+    sorted_centers = sorted(
+        pos for name, pos in sample_centers.items() if name in selected_grids
+    )
+    areas = [
+        (
+            center[0] + rel_bbox[0],
+            center[1] + rel_bbox[1],
+            center[0] + rel_bbox[2],
+            center[1] + rel_bbox[3],
+        )
+        for center in sorted_centers
+    ]
+
+    return areas
+
+def compute_area_size(
+    pos: Dict[str, float],
+    streams: List[Stream],
+    tiles_nx: int,
+    tiles_ny: int,
+    tiling_rng: Dict[str, list],
+    overlap: float,
+) -> Optional[Tuple[float, float]]:
+    """
+    Calculates the requested tiling area size, based on the number of tiles, stream fov and overlap.
+    :param pos: the current position of the stage
+    :param streams: the streams to acquire
+    :param tiles_nx: the number of tiles in the x direction
+    :param tiles_ny: the number of tiles in the y direction
+    :param tiling_rng: the tiling range along x and y axes as (xmin, ymin, xmax, ymax), or (xmin, ymax, xmax, ymin)
+    :param overlap: the overlap between tiles (in percentage)
+    :return: the size (in m) in X and Y. If no area at all, returns None.
+    """
+    # get smallest fov
+    fovs = [get_fov(s) for s in streams]
+    if not fovs:
+        # fall back to a small fov (default)
+        fov = DEFAULT_FOV
+    else:
+        # smallest fov
+        fov = (min(f[0] for f in fovs), min(f[1] for f in fovs))
+
+    # these formulas for w and h have to match the ones used in the 'stitching' module.
+    w = tiles_nx * fov[0] * (1 - overlap)
+    h = tiles_nx * fov[1] * (1 - overlap)
+
+    # Note the area can accept LTRB or LBRT.
+    area = clip_tiling_area_to_range(w, h, pos, tiling_rng)
+    if area is None:
+        # there is no intersection
+        logging.warning(
+            "Couldn't find intersection between stage pos %s and tiling range %s"
+            % (pos, tiling_rng)
+        )
+
+    return area
+
+def get_fov(s: Stream):
+    """Get the field of view of a stream"""
+    try:
+        return s.guessFoV()
+    except (NotImplementedError, AttributeError):
+        raise TypeError("Unsupported Stream %s, it doesn't have a .guessFoV()" % (s,))
+
+def clip_tiling_area_to_range(
+    w: float, h: float, pos: Dict[str, float], tiling_rng: Dict[str, List[float]]
+) -> Optional[Tuple[float]]:
+    """
+    Finds the intersection between the requested tiling area and the tiling range.
+    :param w: the width of the tiling area
+    :param h: the height of the tiling area
+    :param pos: the current position of the stage
+    :param tiling_rng: the tiling range along x and y axes as
```

Comment:
Might be useful to highlight where this tiling range originates from and why do we clip?